app/services/recipe_generator_service.rb

```diff
@@ -6,9 +6,10 @@ class RecipeGeneratorService
   OPENAI_TEMPERATURE = ENV.fetch('OPENAI_TEMPERATURE', 0).to_f
   OPENAI_MODEL = ENV.fetch('OPENAI_MODEL', 'gpt-4')
 
-  def initialize(message, user_id)
+  def initialize(message, user_id, preferences = [])
     @message = message
     @user = User.find(user_id)
+    @preferences = preferences
```

Comment:
You can avoid adding this parameter, since you have user:

```suggestion
    @preferences = user.preferences
```